adview-manager/src/lib.rs

```diff
@@ -211,18 +200,22 @@ pub fn get_unit_html_with_events(
     validators: &Validators,
     no_impression: impl Into<bool>,
 ) -> String {
-    let get_body = |event_type: &str| EventBody {
-        events: vec![Event {
-            event_type: event_type.to_string(),
-            publisher: options.publisher_addr,
-            ad_unit: ad_unit.id,
-            ad_slot: options.market_slot,
-            referrer: "document.referrer".to_string(),
-        }],
-    };
-
     let get_fetch_code = |event_type: &str| -> String {
-        let body = serde_json::to_string(&get_body(event_type))
+        let event = match event_type {
+            "CLICK" => Event::Click {
+                publisher: options.publisher_addr,
+                ad_unit: Some(ad_unit.id),
+                ad_slot: Some(options.market_slot),
+                referrer: Some("document.referrer".to_string()),
+            },
+            _ => Event::Impression {
+                publisher: options.publisher_addr,
+                ad_unit: Some(ad_unit.id),
+                ad_slot: Some(options.market_slot),
+                referrer: Some("document.referrer".to_string()),
+            }
+        };
+        let body = serde_json::to_string(&vec![event])
```

Comment:
As far as I remember the POST .../event takes a body of `{events: [....]}` and not `[....]`. Please double check that and make the necessary adjustments